src/base/Qv2rayLog.cpp

```diff
@@ -0,0 +1,49 @@
+#include "Qv2rayLog.hpp"
+#include "GlobalInstances.hpp"
+
+namespace Qv2ray::base
+{
+    // Forwarded from QvTinyLog
+    static QQueue<QString> __loggerBuffer;
```

Comment:
Seems that `QQueue<>` is not thread-safe.
Maybe you need something like `QMutex` & `QMutexLocker` to make it safe.